tests/integration/pipeline/base.py

```diff
@@ -0,0 +1,122 @@
+        return run_command_with_input(command_list, ("\n".join(inputs) + "\n").encode())
+
+
+class InitIntegBase(PipelineBase):
+    generated_files: List[Path] = []
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        # we need to compare the whole generated template, which is
+        # larger than normal diff size limit
+        cls.maxDiff = None
+
+    def setUp(self) -> None:
+        super().setUp()
+        self.generated_files = []
+
+    def tearDown(self) -> None:
+        for generated_file in self.generated_files:
+            if generated_file.is_dir():
+                shutil.rmtree(generated_file, ignore_errors=True)
+            elif generated_file.exists():
+                generated_file.unlink()
+        super().tearDown()
+
+    def get_init_command_list(
+        self,
+    ):
+        command_list = [self.base_command(), "pipeline", "init"]
+        return command_list
+
+
+class BootstrapIntegBase(PipelineBase):
+    stack_names: List[str]
+
+    @classmethod
+    def setUpClass(cls):
+        cls.cf_client = boto3.client("cloudformation")
+
+    def setUp(self):
+        self.stack_names = []
+        super().setUp()
+
+    def tearDown(self):
+        for stack_name in self.stack_names:
+            self.cf_client.delete_stack(StackName=stack_name)
+        shutil.rmtree(os.path.join(os.getcwd(), ".aws-sam", "pipeline"), ignore_errors=True)
+        super().tearDown()
+
+    def get_bootstrap_command_list(
+        self,
+        no_interactive: bool = False,
+        stage_name: Optional[str] = None,
+        pipeline_user: Optional[str] = None,
+        pipeline_execution_role: Optional[str] = None,
+        cloudformation_execution_role: Optional[str] = None,
+        artifacts_bucket: Optional[str] = None,
+        create_ecr_repo: bool = False,
+        ecr_repo: Optional[str] = None,
+        pipeline_ip_range: Optional[str] = None,
+        no_confirm_changeset: bool = False,
+    ):
+        command_list = [self.base_command(), "pipeline", "bootstrap"]
+
+        if no_interactive:
+            command_list += ["--no-interactive"]
+        if stage_name:
+            command_list += ["--stage-name", stage_name]
+        if pipeline_user:
+            command_list += ["--pipeline-user", pipeline_user]
+        if pipeline_execution_role:
+            command_list += ["--pipeline-execution-role", pipeline_execution_role]
+        if cloudformation_execution_role:
+            command_list += ["--cloudformation-execution-role", cloudformation_execution_role]
+        if artifacts_bucket:
+            command_list += ["--artifacts-bucket", artifacts_bucket]
+        if create_ecr_repo:
+            command_list += ["--create-ecr-repo"]
+        if ecr_repo:
+            command_list += ["--ecr-repo", ecr_repo]
+        if pipeline_ip_range:
+            command_list += ["--pipeline-ip-range", pipeline_ip_range]
+        if no_confirm_changeset:
+            command_list += ["--no-confirm-changeset"]
```

Comment:
I'm fine with that for the purpose of integration test but it would be great if we refactored it.

Reply:
We used to this "batch" if..., like in https://github.com/aws/aws-sam-cli/blob/5cc7680068c820e972d6165a0cccd21677e2a428/tests/integration/deploy/deploy_integ_base.py#L49-L64

Do you have improvement suggestions?